The file AUTHORS has been updated.

mamakancha opened the pull request:
This file was last updated in September 2016. Some more patches have been submitted by new contributors since then. This file now contains all the authors up till 16th March 2017.

mamakancha committed:
The file AUTHORS has been updated.

This file was last updated in September 2016. Some more patches have been submitted by new contributors since then. This file now contains all the authors up till 16th March 2017.

mamakancha commented:
@isuruf  I have removed all the duplicates I saw and now I suppose no duplicates should be there. However, I think I will get the same error as in #12319 on running `./bin/test`

mamakancha commented:
Please let me know what is causing the problem here.

mamakancha commented:
@isuruf All the checks have passed, so I suppose all the duplicates were removed. Kindly review.

isuruf commented:
See this, https://github.com/sympy/sympy/wiki/Adding-Someone-to-AUTHORS#adding-someone-to-authorsaboutus
particularly,
> Whenever a new contributor has code pushed in, they need to be added to the AUTHORS and doc/src/aboutus files

isuruf commented on AUTHORS:
```diff
@@ -543,3 +540,58 @@ Dennis Meckel <meckel@datenschuppen.de>
+Abdullah Javed Nesar <abduljaved1994@gmail.com>
+Lakshya Agrawal <zeeshan.lakshya@gmail.com>
+shruti <shrutishrm512@gmail.com>
+ArifAhmed1995 <arif.ahmed.5.10.1995@gmail.com>
```
This one and Line 548 are the same

isuruf commented on AUTHORS:
```diff
@@ -543,3 +540,58 @@ Dennis Meckel <meckel@datenschuppen.de>
+bsdz <bsdz@users.noreply.github.com>
+Danny Hermes <daniel.j.hermes@gmail.com>
+psqq <pestov-sa@yandex.ru>
+Amit Kumar <aktech@users.noreply.github.com>
```
Duplicate

isuruf commented on AUTHORS:
```diff
@@ -543,3 +540,58 @@ Dennis Meckel <meckel@datenschuppen.de>
+andreo <andrey.torba@gmail.com>
+Flamy Owl <flamyowl@protonmail.ch>
+Yicong Guo <guoyicong100@gmail.com>
+mamakancha <varun.garg03@gmail.com>
```
Update the name, https://github.com/mamakancha

mamakancha replied:
Okay. I missed that instruction. I checked `doc/src/aboutus.rst` , [here](https://github.com/sympy/sympy/blob/master/doc/src/aboutus.rst)I do not see anything section about Authors. What changes are expected there? A new section?

isuruf reviewed (changes requested):
As I said before, you need to go through their github profiles and use the real name, instead of their github handles. Also their contributions need to be added to doc/src/aboutus

isuruf commented:
Ah, sorry, updating `aboutus.rst` is not needed anymore

mamakancha committed:
Most handles replaced with real names and two more duplicates removed.

mamakancha commented:
@isuruf Sorry for the delay. I have changed all the handles with real names of users which I could find on GitHub. Two duplicates as mentioned have also been removed.
Please review.

isuruf commented on AUTHORS:
```diff
@@ -543,3 +539,56 @@ Dennis Meckel <meckel@datenschuppen.de>
+andreo <andrey.torba@gmail.com>
+Flamy Owl <flamyowl@protonmail.ch>
+Yicong Guo <guoyicong100@gmail.com>
+Varun Garg <varun.garg03@gmail.com>
```
To make this automatic, you need to add this to `.mailmap`. See the instructions in that file

mamakancha replied:
So for the people I have changed their handle names to the original ones , I should add the files in the following format:
`Varun Garg <varun.garg03@gmail.com> mamakancha` 
Since I have no bad email?

mamakancha replied:
Also, I see some duplicates in `.mailmap` like:
[https://github.com/sympy/sympy/blob/master/.mailmap#L13](https://github.com/sympy/sympy/blob/master/.mailmap#L13), Is it safe to remove them?

isuruf replied:
that is not a duplicate. if you have a bad name, I think you can just use, `Varun Garg <varun.garg03@gmail.com>`

isuruf commented on AUTHORS:
```diff
@@ -233,7 +233,6 @@ Prasoon Shukla <prasoon92.iitr@gmail.com>
 Stefen Yin <zqyin@ucdavis.edu>
 Thomas Hisch <t.hisch@gmail.com>
 Madeleine Ball <mpball@gmail.com>
-Case Van Horsen <casevh@gmail.com>
```
I don't know why this name was removed. I ran the script locally and it didn't remove this name or the ones below that were removed.

mamakancha replied:
For Line 236, I found duplicate at Line 42 : [here](https://github.com/sympy/sympy/blob/master/AUTHORS#L42)
For Line 477, I found duplicate at Line 393 : [here](https://github.com/sympy/sympy/blob/master/AUTHORS#L393)
For Line 525, I found duplicate at Line 486 : [here](https://github.com/sympy/sympy/blob/master/AUTHORS#L486)
For Line 527., I found duplicate at Line 528 : [here](https://github.com/sympy/sympy/blob/master/AUTHORS#L528) (See email)
I though these duplicates went unnoticed previously. Please let me know if any(or all) of these should be reverted back?

mamakancha replied:
I have added a new commit as well to update `.mailmap` as done accordingly in [Line 255](https://github.com/sympy/sympy/blob/master/.mailmap#L255) and [Line 256](https://github.com/sympy/sympy/blob/master/.mailmap#L256) in the original file

mamakancha replied:
ping @isuruf

isuruf replied:
Thanks. you should add the above to `.mailmap` as well.

mamakancha replied:
Done. :)

mamakancha committed:
`.mailmap` has been updated.

Due to some bad names in the recently added AUTHORS in the previous commits, `.mailmap` also has been updated.

mamakancha committed:
Changes made in AUTHORS and .mailmap

smichr commented on .mailmap:
```diff
@@ -254,3 +254,17 @@ Arihant Parsoya <parsoyaarihant@gmail.com> arihant parsoya <parsoyaarihant@gmail
+Devang Kulshreshtha <devang.kulshreshtha.cse14@itbhu.ac.in> devangkulshreshtha <devang.kulshreshtha.cse14@itbhu.ac.in>
+Mohammad Sadeq Dousti <msdousti@gmail.com> msdousti <msdousti@gmail.com>
+Rohit Rango <rohit.rango@gmail.com> rohitrango <rohit.rango@gmail.com>
+Sergey Pestov <pestov-sa@yandex.ru>	psqq <pestov-sa@yandex.ru>
```
The extra spaces before `psqq` should be removed.

mamakancha replied:
Thanks for pointing out! Done :)

smichr commented on AUTHORS:
```diff
@@ -378,7 +377,7 @@ Eric Miller <emiller42@gmail.com>
 Cody Herbst <cyherbst@gmail.com>
 Nishith Shah <nishithshah.2211@gmail.com>
-AMiT Kumar <dtu.amit@gmail.com>
+Amit Kumar <dtu.amit@gmail.com>
```
We allow abbreviations so perhaps we should allow non-traditional casing if that's how the person represents their name. Did you check with Amit to see if this is his preferred "signature"?

mamakancha replied:
Yes! [here](https://github.com/aktech) is the link to his GitHub profile. From here I figured out that "Amit" would be more appropriate to keep.

smichr replied:
ok

isuruf replied:
@aktech, you changed back to "Amit"?

aktech replied:
@isuruf Yes, I prefer "Amit".

mamakancha committed:
Extra space in Line 261 removed.

mamakancha commented:
@isuruf Is this ready to be merged or some other changes need to be taken care of as well?

asmeurer commented on .mailmap:
```diff
@@ -254,3 +254,17 @@ Arihant Parsoya <parsoyaarihant@gmail.com> arihant parsoya <parsoyaarihant@gmail
+Tanay Agrawal <tanay_agrawal@hotmail.com> tanayag <tanay_agrawal@hotmail.com>
+Mohit Chandra <mohit.chandra@research.iiit.ac.in> mohit3011 <mohit.chandra@research.iiit.ac.in>
+Amit Kumar <dtu.amit@gmail.com> AMiT Kumar <aktech@users.noreply.github.com>
+Case Van Horsen <casevh@gmail.com> Case Van Horsen <casevh@gmail.com>
```
Is this the same name twice? 

I think before Case was not in the git metadata, but now he is, so a special case needs to be added to the authors update script.

asmeurer commented:
I've finished this at https://github.com/sympy/sympy/pull/12834
